src/main/java/com/sapozhnykov/dao/auth/impl/AuthClientDBDaoImpl.java

```diff
@@ -0,0 +1,161 @@
+package com.sapozhnykov.dao.auth.impl;
+
+import com.sapozhnykov.dao.auth.AuthClientDao;
+import com.sapozhnykov.domain.AuthClient;
+
+import java.sql.*;
+import java.util.ArrayList;
+import java.util.List;
+
+public class AuthClientDBDaoImpl implements AuthClientDao {
+    private static final AuthClientDao AUTH_CLIENT_DAO = new AuthClientDBDaoImpl();
+
+    private List<AuthClient> authClients = new ArrayList<>();
+    private static long tempID = -1;
+
+    private static final String DB_URL = "jdbc:h2:tcp://localhost/~/LuxoftShop";
+    private static final String LOGIN = "admin";
+    private static final String PASSWORD = "";
+
+    private AuthClientDBDaoImpl() {
+        try ( Connection connection = DriverManager.getConnection(DB_URL, LOGIN, PASSWORD);
+              PreparedStatement statement = connection.prepareStatement(
+                      "CREATE TABLE IF NOT EXISTS AUTHCLIENT (ID BIGINT PRIMARY KEY AUTO_INCREMENT, USERID BIGINT, PHONE VARCHAR(255), PASSWORD VARCHAR(255))"))
+        {
+            statement.executeUpdate();
+
+            statement.close();
+            connection.close();
+        } catch (SQLException e) {
+            e.printStackTrace();
+        }
+    }
+
+    @Override
+    public AuthClient checkSingIn(String phone, String password) {
+        AuthClient authClient = null;
+
+        try ( Connection connection = DriverManager.getConnection(DB_URL, LOGIN, PASSWORD);
+              PreparedStatement statement = connection.prepareStatement("SELECT * FROM AUTHCLIENT WHERE PHONE=? AND PASSWORD=?;") )
+        {
+            statement.setString(1, phone);
+            statement.setString(2, password);
+
+            try(ResultSet resultSet = statement.executeQuery()) {
+                if(resultSet.next()) {
+                    long authID = resultSet.getLong("ID");
+                    long tempUserID = resultSet.getLong("USERID");
+                    String tempPhone = resultSet.getString("PHONE");
+                    String tempPassword = resultSet.getString("PASSWORD");
+
+                    authClient = new AuthClient(tempUserID, authID, tempPhone, tempPassword);
+                }
+
+
+                resultSet.close();
+            }
+            statement.close();
+            connection.close();
+        } catch (SQLException e) {
+            e.printStackTrace();
+        }
+
+        return authClient;
+    }
+
+    @Override
+    public boolean add(long userId, String phone, String password) {
+        try ( Connection connection = DriverManager.getConnection(DB_URL, LOGIN, PASSWORD);
+              PreparedStatement statement = connection.prepareStatement("INSERT INTO AUTHCLIENT(USERID, PHONE, PASSWORD) VALUES(?, ?, ?)"))
+        {
+            statement.setLong(1, userId);
+            statement.setString(2, phone);
+            statement.setString(3, password);
+            int rows = statement.executeUpdate();
+
+            statement.close();
```

Comment:
WTF? You are already use try-with-resources

Reply:
Fixed. Removed "close" here and in similar Dao

Reply:
I still can see it

Reply:
Fixed. I just missed one